forecasting/src/autogluon/forecasting/predictor/predictor.py

```diff
@@ -0,0 +1,195 @@
+
+        self._learner: AbstractLearner = learner_type(path_context=path, eval_metric=eval_metric, **learner_kwargs)
+        self._learner_type = type(self._learner)
+        self._trainer = None
+
+    def fit(self,
+            train_data,
+            prediction_length,
+            index_column="index_column",
+            time_column="time_column",
+            target_column="target_column",
+            val_data=None,
+            hyperparameters=None,
+            time_limits=None,
+            **kwargs):
+        if self._learner.is_fit:
+            raise AssertionError(
+                'Predictor is already fit! To fit additional models, refer to `predictor.fit_extra`, or create a new '
+                '`Predictor`.')
+
+        self.index_column = index_column
+        self.time_column = time_column
+        self.target_column = target_column
+        kwargs_orig = kwargs.copy()
+        kwargs = self._validate_fit_kwargs(kwargs)
+        kwargs = self._validate_hyperparameter_tune_kwargs(kwargs)
+        verbosity = kwargs.get('verbosity', self.verbosity)
+        set_logger_verbosity(verbosity, logger=logger)
+        hyperparameter_tune = kwargs["hyperparameter_tune"]
+        num_trials = kwargs["num_trials"]
+        search_strategy = kwargs["search_strategy"]
+
+        if verbosity >= 3:
+            logger.log(20, '============ fit kwarg info ============')
+            logger.log(20, 'User Specified kwargs:')
+            logger.log(20, f'{pprint.pformat(kwargs_orig)}')
+            logger.log(20, 'Full kwargs:')
+            logger.log(20, f'{pprint.pformat(kwargs)}')
+            logger.log(20, '========================================')
+
+        freq = kwargs.get("freq", None)
+        set_logger_verbosity(verbosity, logger)
+        if isinstance(train_data, pd.DataFrame):
+            logger.log(30, "Training with dataset in tabular format...")
+            train_data = rebuild_tabular(train_data,
+                                         index_column=index_column,
+                                         target_column=target_column,
+                                         time_column=time_column)
+            logger.log(30, "Finish rebuilding the data, showing the top five rows.")
+            logger.log(30, train_data.head())
+            if val_data is None:
+                logger.log(30, "Validation data is None, will do auto splitting...")
+                train_data, val_data = train_test_split_dataframe(train_data, prediction_length)
+            else:
+                val_data = rebuild_tabular(val_data,
+                                           index_column=index_column,
+                                           target_column=target_column,
+                                           time_column=time_column)
+            train_data = TimeSeriesDataset(train_data, index_column=index_column)
+            freq = train_data.get_freq()
+            if val_data is not None:
```

Comment:
Is `if val_data is not None:` used for double checking val_data? It looks like you already assigned rebuild_tabular output to valdata.

Reply:
It seems that this check is redundant and I have removed it.

Reply:
> Is `if val_data is not None:` used for double checking val_data? It looks like you already assigned rebuild_tabular output to valdata.

It seems that this check is redundant and I have removed it.